packages/api/src/repository.js

```diff
@@ -78,6 +78,69 @@ class Repository {
     const document = client.doc(path);
     return document.delete();
   }
+
+  async deleteQueryBatch (query, resolve) {
+    const snapshot = await query.get();
+
+    const batchSize = snapshot.size;
+    if (batchSize === 0) {
+    // When there are no documents left, we are done
+      resolve();
+      return;
+    }
+
+    // Delete documents in a batch
+    const batch = client.batch();
+    snapshot.docs.forEach((doc) => {
+      batch.delete(doc.ref);
+    });
+    await batch.commit();
+
+    // Recurse on the next process tick, to avoid
+    // exploding the stack.
+    process.nextTick(() => this.deleteQueryBatch(query, resolve));
+  }
+
+  async deleteCollection (collectionPath, batchSize) {
+    const collectionRef = client.collection(collectionPath);
+    const query = collectionRef.limit(batchSize);
+    return new Promise((resolve, reject) => {
+      this.deleteQueryBatch(query, resolve).catch(reject);
+    });
+  }
+
+  async deleteExpiredSessions () {
+    console.log('Deleting Expired Sessions'); // Not deleted because it's useful to see when the crob job runs.
+    const collectionRef = client.collection('express-sessions');
+    const snapshot = await collectionRef.get();
+
+    // Delete documents in a batch
+    let batch = client.batch();
+    let numSeen = 0;
+    await snapshot.docs.forEach(async (doc) => {
+      try {
+        const data = await JSON.parse(doc.data().data);
+        console.log('DATA: ' + data);
+        const expiration = data.expires;
+        if (expiration == null || moment().isAfter(moment(expiration))) {
+          await batch.delete(doc.ref);
+          console.log('DELETING');
+          numSeen++;
+        }
+      } catch (err) {
+        await batch.delete(doc.ref);
+        console.log('DELETING');
+        numSeen++;
+      }
+      if (numSeen >= 300) {
+        console.log('COMMITTING');
+        await batch.commit();
+        batch = client.batch();
```

Comment:
I cant quite tell from looking at it, but is it ever possible you would exceed the 500 limit for batch operations?

Reply:
I had some trouble with this bit of code, it does seem to struggle when there are a lot of sessions to delete. It works for small amounts, which is all that's needed in production since it will be constantly running, so the amount of sessions will never get too big, but I plan to return to this later and test/fix it up more.

Reply:
I wonder if maybe you could get an array for all the docs, and instead of sifting one-by-one, instead using .filter() to get a new array of expired docs. THen you could split these up into the appropriate groups (500?) and then delete those. 

As a further note, I would recommend against doing the same thing in a catch block as your try block. It looks like you're trying to build a retry but that is not the preferred way of doing that. For now, I would just catch the error, and somehow make sure you're putting that doc back in a list to look over, rather than immediately trying to delete again.